lib/sorbet-rails/model_plugins/active_record_attribute.rb

```diff
@@ -107,4 +112,41 @@ def active_record_type_to_sorbet_type(klass)
       end
     end
   end
+
+  # True if this column is "time zone aware", which means it'll be converted on
+  # access from its original class (e.g. `DateTime`) to something with better
+  # support for time zones (usually `ActiveSupport::TimeWithZone`)
+  sig do
+    params(column_def: T.untyped, cast_type: T.untyped).returns(T::Boolean)
+  end
+  def time_zone_aware_column?(column_def, cast_type)
+    # this private class method returns true if the attribute should be "time
+    # zone aware"; it takes into account various global and model-specific
+    # configuration options as described here:
+    # https://api.rubyonrails.org/classes/ActiveRecord/Timestamp.html
+    #
+    # although it's private, it's better this than trying to reimplement the "is
+    # this attribute tz aware?" logic ourselves
+    @model_class.send(
+      :create_time_zone_conversion_attribute?,
```

Comment:
Although private, this method has been in place for **10 years**, so as private methods go, it seems quite stable!

https://github.com/rails/rails/blame/55cc16f502399b69051e4ff5432569bbea45e9c0/activerecord/lib/active_record/attribute_methods/time_zone_conversion.rb